src/components/Identity/index.js

```diff
@@ -37,6 +39,17 @@ const createIdentity = ({ config, logger, cookie }) => {
       },
       // Waiting for opt-in because we'll be reading the ECID from a cookie
       onBeforeEvent(event) {
+        const nowInHours = Math.round(
```

Comment:
I think we should run this logic after we've received the opt-in consent, especially that we are reading the timestamp from the cookie.